pkg/apis/core/types_shoot.go

```diff
@@ -497,6 +497,33 @@ type KubeAPIServerConfig struct {
 	EnableAnonymousAuthentication *bool
 	// EventTTL controls the amount of time to retain events.
 	EventTTL *metav1.Duration
+	// AccessControl provides authorization mechanisms for Shoot-API-Server.
+	AccessControl *AccessControl
+}
+
+// AccessControl provides authorization mechanisms for Shoot-API-Server.
+// Note: The schema (incl. child structs) and documentation resembles istio's AuthorizationPolicy.
+type AccessControl struct {
+	// The action to take on the source of request.
+	Action *AuthorizationAction
+	// Origin of request to run defined authorization action against.
+	Source *Source
+}
+
+type AuthorizationAction string
```

Comment:
several doc strings for exported types or constants are missing (same in `v1beta1/types_shoot.go`)